### PR TITLE
Remove deprecated modules (asyncio, sockets, ftpclient)

### PR DESCRIPTION
**Pull request by narimiran**
These three modules (`asyncio`, `sockets`, `ftpclient`) were deprecated since v0.10.2 and v0.11.3.

Some tests imported them, so now those imports are replaced with the modules which superseded those (`asyncnet`, `asyncdispatch`, `nativesockets`, `net`).

This also removes `asyncdispatch.recvLine` proc, which was deprecated since v0.15.0. (Replaced by `asyncnet.recvLine` in the test)

One test (`texplicit_dummy_closure.nim`) had to be disabled, as I couldn't find the easy way to make it work without `asyncio` and `sockets`.

### CODE REVIEW

File: lib/pure/asyncftpclient.nim [hunk 1]
```diff
@@ -75,12 +75,57 @@
...
 
-import asyncdispatch, asyncnet, strutils, parseutils, os, times
-
-from ftpclient import FtpBaseObj, ReplyError, FtpEvent
+import asyncdispatch, asyncnet, nativesockets, strutils, parseutils, os, times
 from net import BufferSize
 
 type
+  FtpBase*[SockType] = ref FtpBaseObj[SockType]
+  FtpBaseObj*[SockType] = object
```
Comment (dom96) [inline lib/pure/asyncftpclient.nim]: I would get rid of the non-async variant and simply move the fields into the ``AsyncFtpClient``. You can also remove the ``*Obj`` variants I think.

File: tests/async/tasyncawait.nim [hunk 2]
```diff
@@ -12,20 +12,21 @@ const
...
-    var sock = newAsyncNativeSocket()
+  for i in 0 ..< swarmSize:
+    var sock = createAsyncNativeSocket()
 
     await connect(sock, "localhost", port)
     await sendMessages(sock)
     closeSocket(sock)
 
 proc readMessages(client: AsyncFD) {.async.} =
+  var a = newAsyncSocket(client)
```
Comment (dom96) [inline tests/async/tasyncawait.nim]: please name this something more descriptive like `sockObj`.

Comment (dom96) [reply tests/async/tasyncawait.nim]: Also, a comment might help, took me a few seconds to understand why you were creating a new socket. You're actually wrapping the AsyncFd into a AsyncSocket object.

File: tests/closure/texplicit_dummy_closure.nim [hunk 3]
```diff
@@ -1,3 +1,6 @@
+discard """
+  disabled: true
+"""
 
 # This is a regression of the new lambda lifting; detected by Aporia
```
Comment (dom96) [inline tests/closure/texplicit_dummy_closure.nim]: Oh nooo. This means Aporia won't compile at all now. IIRC it still uses asyncio D:

Not something I expect you to fix, just makes me sad :)

Comment (Araq) [reply tests/closure/texplicit_dummy_closure.nim]: Maybe update Aporia then?

**Pull request merged**